commcare_connect/reports/helpers.py

```diff
@@ -1,101 +1,170 @@
+        .values_list("visit_date")
+        .order_by("-visit_date")[:1]
+    )
+    time_to_payment = ExpressionWrapper(F("payment_date") - Subquery(max_visit_date), output_field=DurationField())
     visit_data = (
         CompletedWork.objects.annotate(work_date=Max("uservisit__visit_date"))
         .filter(
             **filter_kwargs,
-            opportunity_access__opportunity__is_test=False,
             status=CompletedWorkStatus.approved,
             work_date__gte=start_date,
             work_date__lt=end_date,
         )
-        .select_related("opportunity_access__opportunity__delivery_type")
+        .values("opportunity_access__opportunity__delivery_type__name")
+        .annotate(
+            users=Count("opportunity_access__user_id", distinct=True),
+            service_count=Sum("saved_approved_count", default=0),
+            flw_amount_earned=Sum("saved_payment_accrued", default=0),
+            nm_amount_earned=Sum(
+                F("saved_approved_count")
+                * F("opportunity_access__opportunity__managedopportunity__org_pay_per_visit"),
+                default=0,
+            ),
+            avg_time_to_payment=Avg(time_to_payment, default=timedelta(days=0)),
+            max_time_to_payment=Max(time_to_payment, default=timedelta(days=0)),
+        )
     )
-    for v in visit_data:
-        delivery_type_name = "All"
-        if group_by_delivery_type and v.opportunity_access.opportunity.delivery_type:
-            delivery_type_name = v.opportunity_access.opportunity.delivery_type.name
-
-        user_set[delivery_type_name].add(v.opportunity_access.user_id)
-        beneficiary_set[delivery_type_name].add(v.entity_id)
-        service_count[delivery_type_name] += v.saved_approved_count
 
     payment_query = Payment.objects.filter(
-        **filter_kwargs,
-        opportunity_access__opportunity__is_test=False,
         date_paid__gte=start_date,
         date_paid__lt=end_date,
     )
 
-    approved_payment_data = (
-        payment_query.filter(confirmed=True)
-        .values("opportunity_access__opportunity__delivery_type__name")
+    user_count_data = {
+        item["opportunity_access__opportunity__delivery_type__name"]: item["users"] for item in visit_data
+    }
+    service_count_data = {
+        item["opportunity_access__opportunity__delivery_type__name"]: item["service_count"] for item in visit_data
+    }
+    avg_time_to_payment_data = {
+        item["opportunity_access__opportunity__delivery_type__name"]: item["avg_time_to_payment"]
+        for item in visit_data
+    }
+    max_time_to_payment_data = {
+        item["opportunity_access__opportunity__delivery_type__name"]: item["max_time_to_payment"]
+        for item in visit_data
+    }
+    flw_amount_earned_data = {
+        item["opportunity_access__opportunity__delivery_type__name"]: item["flw_amount_earned"] for item in visit_data
+    }
+    nm_amount_earned_data = {
```

Comment:
I think this and the other NM amounts should be the sum of this and the FLW amounts (the nm amount includes the part they pass through to the FLW), you can see the other dashboard for an example

Reply:
Fixed. https://github.com/dimagi/commcare-connect/pull/487/commits/3b286518bd5c62c1d38727dc6d5bbb2c73c1a350